pkg/neotest/basic.go

```diff
@@ -258,29 +257,30 @@ func (e *Executor) EnsureGASBalance(t testing.TB, acc util.Uint160, isOk func(ba
 }
 
 // NewDeployTx returns a new deployment tx for the contract signed by the committee.
-func (e *Executor) NewDeployTx(t testing.TB, bc *core.Blockchain, c *Contract, data any) *transaction.Transaction {
-	return NewDeployTxBy(t, bc, e.Validator, c, data)
+func (e *Executor) NewDeployTx(t testing.TB, c *Contract, data any) *transaction.Transaction {
+	return e.NewDeployTxBy(t, e.Validator, c, data)
 }
 
 // NewDeployTxBy returns a new deployment tx for the contract signed by the specified signer.
-func NewDeployTxBy(t testing.TB, bc *core.Blockchain, signer Signer, c *Contract, data any) *transaction.Transaction {
+func (e *Executor) NewDeployTxBy(t testing.TB, signer Signer, c *Contract, data any) *transaction.Transaction {
 	rawManifest, err := json.Marshal(c.Manifest)
 	require.NoError(t, err)
 
 	neb, err := c.NEF.Bytes()
 	require.NoError(t, err)
 
-	script, err := smartcontract.CreateCallScript(bc.ManagementContractHash(), "deploy", neb, rawManifest, data)
+	script, err := smartcontract.CreateCallScript(e.Chain.ManagementContractHash(), "deploy", neb, rawManifest, data)
 	require.NoError(t, err)
 
-	tx := transaction.New(script, 100*native.GASFactor)
+	tx := transaction.New(script, 0)
```

Comment:
RPC GAS limit problem?

Reply:
Not until https://github.com/nspcc-dev/neo-go/issues/3245 implemented. And when it's implemented, we can easily detect insufficient GAS error and handle it on the Executor side by replacing tx's system fee with some constant value. I'll add a comment to #3245.

And right now AddSystemFee uses VM with unlimited GAS.

Reply:
> And right now AddSystemFee uses VM with unlimited GAS.

OK, OK.